composer.json

```diff
@@ -12,7 +12,8 @@
     ],
     "require": {
         "php":          "^7.1",
-        "symfony/yaml": "^3.0"
+        "symfony/yaml": "^3.0",
+        "composer/semver": "^1.4"
```

Comment:
I've run the patch locally, and it seems to assume only semver versioning because of this dependency. Versions of packages in here are not required to follow SemVer, so some assumptions in here are broken because of that (most of those assumptions are hidden by the fact that some tests were removed)